citrination_client/views/client.py

```diff
@@ -46,13 +51,19 @@ def create(self, configuration, name, description):
             'v1/data_views', data, failure_message=failure_message))
         data_view_id = result['data']['id']
 
+        if not is_async:
+            self._wait_for(
+                "Predict services ready",
```

Comment:
It's not clear that me that this is desirable or obvious behavior. It makes me wonder if the view is actually finished while model reports are still being built. What happens if the view can't train any models? Is that a timeout?

Reply:
Agreed that this is a little prescriptive in kind of an odd way. It does align with lots of workflows we've seen in data science though -- often, you'll derisk a view by building it and watching to see what happens in the UI (or polling status in API). But once that view works / trains, you may want to build ~10 other views that are slightly different and you have high confidence will work (e.g. slightly different ML configs w/ varying inputs/outputs), and typically these subsequent views will work fine -- so this is just a convenience flag so that when the function returns, you know that you can load the view and look at model reports.

Reply:
Would it significantly change the workflow to give users a utility, so that they can be explicit about what behavior they want? Something like:

```python
models_client.retrain(view_id)
models_client.wait_for_retrain(view_id)
print("retrain finished - updated models are ready")

models_client.wait_for_model_reports(view_id)
print("updated model reports ready")
```

Reply:
A lot of my motivation for this comes from writing the SL demo, where `upload`, `create_view`, `retrain` and `design` are happening in a rapid loop, and where function calls need to be synchronous (i.e. next cell or next iteration shouldn't run until previous process is done - very similar to a workflow-test).

Also, I agree with what @eddotman is saying - this would help mitigate the part of DS workflows where we create a view and then stare at the web UI until reports are ready, or upload a dataset then watch ingest progress on the Web UI. But we're smart enough to add our own `waits`, for the most part I think. This is more for non-power-users and community folks who are just getting started.

I think, with respect to both of the above use cases, we're trying to address the following:
Does it help the user to return a successful `upload_result` when the user can't actually do anything with those uploaded files yet? I understand the usefulness internally, but I can't think of a user workflow that doesn't require [this entire series of checks] before doing anything else(https://github.com/CitrineInformatics/citrination-workflow-tests/blob/master/workflow_tests/stress/heavy_correctness_tests/base/data_view_workflow.py#L66-L97).

Similarly, I'm struggling to think of a workflow in which returning a `data_view_id` after a `create()` call is helpful to the user if the services are ultimately going to fail.

Having written that, though, I see the folly of `_waiting_on` predict services to be _ready_, when they might just fail outright.

Reply:
there's two ways this could be implemented:
1) *build the `_wait` into the resource instantiation*: don't return anything from the instantiating calls (`upload` or `create`) until the resource is ready to be used
2) *build the `_wait` into the individual resource usage functions*: allow user to call `retrain`, `predict`, `design` on a View, or try to `get_available_columns` from a Dataset, but have those functions `wait` until they can be used successfully. I know that this is already kind of baked into predict requests and design runs.

Reply:
To be clear, I'm still not blocking this, but for the sake of the dialogue:

I think where I'm coming from is just a general software bias towards simple, atomic functions as the basic units of work. If you want to support an SL workflow that has some assumptions about use case built in, then I'd create a separate package to support that workflow (it could even be a `SequentialLearningClient` that uses the atomic API + sugar). This way, you retain the flexibility and coherence of the base API while also making your users' lives easier for common workflows.

Reply:
I appreciate that, and I don't want to bloat PyCC unnecessarily. An `SL_Client` is a great idea, although it would take longer to put together (and I'm about to go on paternity leave). That would also be a good place to integrate some of the other design wrappers we have.

I'm OK closing without merging, but using what we've started here to build an SL Client in the future. What do you think, @eddotman ?

Reply:
Agreed that an SLClient is a great path forward. I do think that if we close this w/o merging, there is a high risk of forgetting to come back to it. 

Can we leave this PR open (probably for a while) and block closing it on the creation of a `SLClient` some time later this summer?